spacy/language.py

```diff
@@ -777,8 +777,8 @@ def add_pipe(
             component directly before.
         after (Union[str, int]): Name or index of the component to insert new
             component directly after.
-        first (bool): If True, insert component first in the pipeline.
-        last (bool): If True, insert component last in the pipeline.
+        first (True or None): If True, insert component first in the pipeline.
```

Comment:
Wouldn't this be more consistent?
```suggestion
        first (Literal[True]): If True, insert component first in the pipeline.
```

Reply:
In terms of consistency, you're right that throughout the code base we use `Optional` much more often than `or None` in the doc strings, though there are a handful cases of the latter. Happy to change it, though in that case it should be `Optional[Literal[True]]`.

Reply:
Fixed the suggestion. Up to you :slightly_smiling_face: